In [2]:
import pandas as pd
from sklearn.model_selection import train_test_split

#df = pd.read_csv("/content/Advertising.csv", index_col=0)

In [ ]:
df


TV  radio  newspaper  sales
1    230.1   37.8       69.2   22.1
2     44.5   39.3       45.1   10.4
3     17.2   45.9       69.3    9.3
4    151.5   41.3       58.5   18.5
5    180.8   10.8       58.4   12.9
..     ...    ...        ...    ...
196   38.2    3.7       13.8    7.6
197   94.2    4.9        8.1    9.7
198  177.0    9.3        6.4   12.8
199  283.6   42.0       66.2   25.5
200  232.1    8.6        8.7   13.4

[200 rows x 4 columns]

In [ ]:
from sklearn.preprocessing import LabelEncoder

train, test = train_test_split(df, test_size=0.3, random_state=123)

x_train = train[['TV','radio','newspaper']]
y_train = train['sales']

x_test = test[['TV','radio','newspaper']]
y_test = test['sales']


In [ ]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_validate
import matplotlib.pyplot as plt

model = LinearRegression()
model.fit(x_train, y_train)
coeficientes = model.coef_
interseccion = model.intercept_

print(coeficientes)
print(interseccion)

[0.04663016 0.18406264 0.00069216]
2.780986354357207


- ¿qué significa el valor del punto de corte para el problema que se soluciona?

El valor del punto de corte es muy importante pues es este el que indica el valor base de la variable dependiente cuando no hay influencia de las variables independientes (Como si las otras fueran 0).

En este ejemplo, en un modelo de regresión que predice las ventas basado en tv, pariodico y la radio. El punto de corte podría representar el valor de las ventas con caro inversion en publicidad en los medios como tv, radio y periodico.

- ¿qué significa el valor de cada coeficiente  para el problema que se soluciona? cómo influyen en la variable a predecir?

Los coeficientes en un modelo de regresión lineal tienen la funcion de mostrar la magnitud y dirección de una relación entre cada variable independiente y la variable dependiente. Cada coeficiente indica cuánto cambia la variable dependiente cuando la variable independiente determinada se incrementa por 1, mientras que todas las demás variables se dejan constantes.

En este ejemplo, si en el mismo modelo de regresión de ventas, el coeficiente para tv es 0.04663016, significa que, en promedio, por cada unidad adicional de inversion en tv, el valor de las ventas aumenta en 0.04663016, manteniendo constantes otras variables como el periodico y radio.

Por lo tanto, el valor de cada coeficiente es crucial para entender cómo influyen las variables independientes en la variable dependiente y cómo se puede usar el modelo para hacer predicciones sobre la variable dependiente en función de los valores de las variables independientes.

In [ ]:
#Tarea
#Utilizar el greman credit para hacer lo de la clase pasada (importancia de bariables con el arbol de decision)
#Volver categoricas a numericas (Getdannys)

In [3]:
df = pd.read_csv("/content/Credit.csv", index_col=0)

In [4]:
df


Limit  Rating  Cards  Age  Education  Own Student Married Region  \
Income                                                                      
14.891    3606     283      2   34         11   No      No     Yes  South   
106.025   6645     483      3   82         15  Yes     Yes     Yes   West   
104.593   7075     514      4   71         11   No      No      No   West   
148.924   9504     681      3   36         11  Yes      No      No   West   
55.882    4897     357      2   68         16   No      No     Yes  South   
...        ...     ...    ...  ...        ...  ...     ...     ...    ...   
12.096    4100     307      3   32         13   No      No     Yes  South   
13.364    3838     296      5   65         17   No      No      No   East   
57.872    4171     321      5   67         12  Yes      No     Yes  South   
37.728    2525     192      1   44         13   No      No     Yes  South   
18.701    5524     415      5   64          7  Yes      No      No   West   

         Balance  
Income            
14.891       333  
106.025      903  
104.593      580  
148.924      964  
55.882       331  
...          ...  
12.096       560  
13.364       480  
57.872       138  
37.728         0  
18.701       966  

[400 rows x 10 columns]

In [5]:
df = pd.get_dummies(df)
df

Limit  Rating  Cards  Age  Education  Balance  Own_No  Own_Yes  \
Income                                                                    
14.891    3606     283      2   34         11      333    True    False   
106.025   6645     483      3   82         15      903   False     True   
104.593   7075     514      4   71         11      580    True    False   
148.924   9504     681      3   36         11      964   False     True   
55.882    4897     357      2   68         16      331    True    False   
...        ...     ...    ...  ...        ...      ...     ...      ...   
12.096    4100     307      3   32         13      560    True    False   
13.364    3838     296      5   65         17      480    True    False   
57.872    4171     321      5   67         12      138   False     True   
37.728    2525     192      1   44         13        0    True    False   
18.701    5524     415      5   64          7      966   False     True   

         Student_No  Student_Yes  Married_No  Married_Yes  Region_East  \
Income                                                                   
14.891         True        False       False         True        False   
106.025       False         True       False         True        False   
104.593        True        False        True        False        False   
148.924        True        False        True        False        False   
55.882         True        False       False         True        False   
...             ...          ...         ...          ...          ...   
12.096         True        False       False         True        False   
13.364         True        False        True        False         True   
57.872         True        False       False         True        False   
37.728         True        False       False         True        False   
18.701         True        False        True        False        False   

         Region_South  Region_West  
Income                              
14.891           True        False  
106.025         False         True  
104.593         False         True  
148.924         False         True  
55.882           True        False  
...               ...          ...  
12.096           True        False  
13.364          False        False  
57.872           True        False  
37.728           True        False  
18.701          False         True  

[400 rows x 15 columns]

In [6]:
df.reset_index(inplace=True)
df

Income  Limit  Rating  Cards  Age  Education  Balance  Own_No  Own_Yes  \
0     14.891   3606     283      2   34         11      333    True    False   
1    106.025   6645     483      3   82         15      903   False     True   
2    104.593   7075     514      4   71         11      580    True    False   
3    148.924   9504     681      3   36         11      964   False     True   
4     55.882   4897     357      2   68         16      331    True    False   
..       ...    ...     ...    ...  ...        ...      ...     ...      ...   
395   12.096   4100     307      3   32         13      560    True    False   
396   13.364   3838     296      5   65         17      480    True    False   
397   57.872   4171     321      5   67         12      138   False     True   
398   37.728   2525     192      1   44         13        0    True    False   
399   18.701   5524     415      5   64          7      966   False     True   

     Student_No  Student_Yes  Married_No  Married_Yes  Region_East  \
0          True        False       False         True        False   
1         False         True       False         True        False   
2          True        False        True        False        False   
3          True        False        True        False        False   
4          True        False       False         True        False   
..          ...          ...         ...          ...          ...   
395        True        False       False         True        False   
396        True        False        True        False         True   
397        True        False       False         True        False   
398        True        False       False         True        False   
399        True        False        True        False        False   

     Region_South  Region_West  
0            True        False  
1           False         True  
2           False         True  
3           False         True  
4            True        False  
..            ...          ...  
395          True        False  
396         False        False  
397          True        False  
398          True        False  
399         False         True  

[400 rows x 16 columns]

In [7]:
from sklearn.preprocessing import LabelEncoder

train, test = train_test_split(df, test_size=0.3, random_state=123)

x_train = train[['Limit',	'Rating',	'Cards',	'Age',	'Education',	'Balance',	'Own_No',	'Own_Yes',	'Student_No',	'Student_Yes',	'Married_No',	'Married_Yes',	'Region_East',	'Region_South',	'Region_West']]
y_train = train['Income']

x_test = test[['Limit',	'Rating',	'Cards',	'Age',	'Education',	'Balance',	'Own_No',	'Own_Yes',	'Student_No',	'Student_Yes',	'Married_No',	'Married_Yes',	'Region_East',	'Region_South',	'Region_West']]
y_test = test['Income']


In [8]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_validate
import matplotlib.pyplot as plt

model = LinearRegression()
model.fit(x_train, y_train)
coeficientes = model.coef_
interseccion = model.intercept_

print(coeficientes)
print(interseccion)

[  0.02322377   0.08580968   1.5032362   -0.02528474  -0.07606564
  -0.09549758   0.92972549  -0.92972549 -20.18513368  20.18513368
   0.97594889  -0.97594889  -1.50788156   0.29610012   1.21178144]
-31.843708035363704
